### ISSUE TITLE
[Screen Reader - Power Apps Admin - Select a template] Narrator in scan mode announcing the incorrect information of the column header associate with the 'Abcd' cell on the table Under Select a template.

### ISSUE THREAD

**Issue by sonimishra** [OfficeDev/office-ui-fabric-react#6831]
### Bug Report 
 
User Experience:
If, the Screen Reader is announcing incorrect information of the column header associated with the cell in a table, then the users who depends upon Screen Reader for navigation in a table might get confuse and does not able to get proper information about the associated column header with the cell.

MAS Ref:
MAS 4.2.3 - Row, Column, and Headers (33C)

Check out Keros! - Identify accessibility bugs before check-in and make bug fixing faster and easier.  
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (Yes, No)
 
Requested priority: (Blocking, High, Normal, Low) 
Blocking
 
Products/sites affected: (if applicable)
 
#### Describe the issue:
 Pre-Requisite:
1. Turn on narrator by pressing Ctrl + window + ENTER combination.

Repro Steps:
1. Navigate to https://admin.powerplatform.microsoft.com/ext/DataIntegration

  and login with intusr.
2. Navigate to the Data Integration menu item on the left bar and select it.
3. Navigate to "+New Project" on the top right corner and select it.
4.New Data integration project screen with Connection tab item selected opens.
5. Navigate on the connection radio button and select it.
6. Navigate to Next button on bottom and select it.
7. Select a template screen appears. Navigate with the Narrator in scan mode on the table and reach 'Abcd' Cell.
8. Verify whether the Narrator in scan mode announcing the correct information of the column header associated with the cell or not.

#### Actual behavior:
When user navigates with the Narrator in scan mode on the table and reach 'Abcd' cell, then the narrator announce the incorrect column header information i.e ''Column header Description, Abcd''.

 #### Expected behavior:
Whenever, user navigates with the Narrator in scan mode on the table and reach 'Abcd' cell, then the narrator should announce the correct column header information i.e ''Column header Name, Abcd''.
 
 Note:
1. This issue does not exist with Chrome + JAWS in cursor mode, as JAWS is not announcing the column header information.

**Comment by cliffkoh** [comment]
Thanks, I am able to repro this issue.

**Comment by cliffkoh** [comment]
Although I can repro the issue, I was not able to identify what was incorrect with the DetailsList.

![image](https://user-images.githubusercontent.com/1003246/47550205-5ca71700-d8b3-11e8-8460-3f543cafda7e.png)

Despite seemingly correct markup, it reads the column header as "Value" instead. Is this potentially a Narrator bug? 

A secondary shield might want to validate the my triage to see if I missed something, and either follow up with a fix or report a bug with the Narrator team.

(cc: @cschlechty)
